app/jobs/create_virtual_objects_job.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+##
+# job to create virtual objects
+class CreateVirtualObjectsJob < GenericJob
+  queue_as :default
+
+  ##
+  # A job that creates virtual objects
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [String] :create_virtual_objects CSV string
+  def perform(bulk_action_id, params)
```

Comment:
This method is indeed gnarly, even gnarlier than it was before I renamed the job. If I have time, I'd like to spend a couple hours refactoring it. I can also ticket this for later.

Reply:
refactoring this as a separate ticket seems like a great approach.  Let's get 'er in front of Andrew asap.  It would be so cool to get this deployed!

Reply:
@ndushay roger that!